pkg/txn/storage/tae/storage.go

```diff
@@ -26,71 +35,144 @@ import (
 	"github.com/matrixorigin/matrixone/pkg/txn/storage"
 )
 
-type Storage struct {
+type TAEStorage struct {
 	shard     metadata.DNShard
 	logClient logservice.Client
 	fs        fileservice.FileService
 	clock     clock.Clock
+	tae       *db.DB
+	txns      struct {
+		sync.Mutex
+		idMap map[string]txnif.AsyncTxn
+	}
 }
 
-func New(
+func openTAE(targetDir string) (*db.DB, error) {
+	mask := syscall.Umask(0)
+	if err := os.MkdirAll(targetDir, os.FileMode(0755)); err != nil {
+		syscall.Umask(mask)
+		logutil.Infof("Recreate dir error:%v\n", err)
+		return nil, err
+	}
+	syscall.Umask(mask)
+	tae, err := db.Open(targetDir+"/tae", nil)
+	if err != nil {
+		logutil.Infof("Open tae failed. error:%v", err)
+		return nil, err
+	}
+	return tae, nil
+}
+
+func NewTAEStorage(
 	shard metadata.DNShard,
 	logClient logservice.Client,
 	fs fileservice.FileService,
 	clock clock.Clock,
-) (*Storage, error) {
-
-	return &Storage{
+) (*TAEStorage, error) {
+	//open tae, just for test.
+	dir := "./store"
+	engine, err := openTAE(dir)
+	if err != nil {
+		return nil, err
+	}
+	storage := &TAEStorage{
 		shard:     shard,
 		logClient: logClient,
 		fs:        fs,
 		clock:     clock,
-	}, nil
+		tae:       engine,
+	}
+	storage.txns.idMap = make(map[string]txnif.AsyncTxn)
+	return storage, nil
 }
 
-var _ storage.TxnStorage = new(Storage)
+var _ storage.TxnStorage = new(TAEStorage)
+
+func (s *TAEStorage) getOrCreate(meta txn.TxnMeta) (txn txnif.AsyncTxn, err error) {
+	s.txns.Lock()
+	defer s.txns.Unlock()
+	txn, ok := s.txns.idMap[string(meta.ID)]
+	if !ok {
+		//TODO::use StartTxnWithMeta, instead of StartTxn.
+		txn, err = s.tae.StartTxn(nil)
+	}
+	return
+}
 
-// Close implements storage.TxnStorage
-func (*Storage) Close(ctx context.Context) error {
-	panic("unimplemented")
+func (s *TAEStorage) get(meta txn.TxnMeta) (txn txnif.AsyncTxn, err error) {
+	s.txns.Lock()
+	defer s.txns.Unlock()
+	txn, ok := s.txns.idMap[string(meta.ID)]
+	if !ok {
+		return nil, moerr.NewMissingTxn()
+	}
+	return
 }
 
-// Commit implements storage.TxnStorage
-func (*Storage) Commit(ctx context.Context, txnMeta txn.TxnMeta) error {
+// Close implements storage.TxnTAEStorage
+func (s *TAEStorage) Close(ctx context.Context) error {
 	panic("unimplemented")
 }
 
-// Committing implements storage.TxnStorage
-func (*Storage) Committing(ctx context.Context, txnMeta txn.TxnMeta) error {
-	panic("unimplemented")
+// Commit implements storage.TxnTAEStorage
+func (s *TAEStorage) Commit(ctx context.Context, txnMeta txn.TxnMeta) error {
+	txnHandle, err := s.get(txnMeta)
+	if err != nil {
+		return err
+	}
+	return txnHandle.Commit()
+
 }
 
-// Destroy implements storage.TxnStorage
-func (*Storage) Destroy(ctx context.Context) error {
+// Committing implements storage.TxnTAEStorage
+func (s *TAEStorage) Committing(ctx context.Context, txnMeta txn.TxnMeta) error {
```

Comment:
This is fine -- but we prefer people to use panic(moerr.NewNYI("what is not implemented"))

Reply:
ok